File: service/app/src/test/java/io/miragon/blueprint/adapter/outbound/contract/LoggingContractAdapterTest.java
package io.miragon.blueprint.adapter.outbound.contract;

import static io.miragon.blueprint.domain.leasing.TestObjectBuilder.testLeasingApplication;
import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatCode;

import io.miragon.blueprint.domain.leasing.ContractId;
import org.junit.jupiter.api.Test;

class LoggingContractAdapterTest {

    private final LoggingContractAdapter underTest = new LoggingContractAdapter();

    @Test
    void issueContractMintsAFreshContractId() {
        // given: an application / when: a contract is issued twice
        ContractId first = underTest.issueContract(testLeasingApplication().build().id());
        ContractId second = underTest.issueContract(testLeasingApplication().build().id());
        // then: each issue gets its own reference
        assertThat(first.value()).startsWith("CONTRACT-");
        assertThat(first).isNotEqualTo(second);
    }

    @Test
    void revokeContractLogsWithoutError() {
        // given: an issued contract / when-then: revoking it runs without error
        assertThatCode(() -> underTest.revokeContract(new ContractId("CONTRACT-1"))).doesNotThrowAnyException();
    }
}

File: service/app/src/test/kotlin/io/miragon/blueprint/adapter/outbound/contract/LoggingContractAdapterTest.kt
package io.miragon.blueprint.adapter.outbound.contract

import io.miragon.blueprint.domain.leasing.ContractId
import io.miragon.blueprint.domain.leasing.testLeasingApplication
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatCode
import org.junit.jupiter.api.Test

class LoggingContractAdapterTest {

    private val underTest = LoggingContractAdapter()

    @Test
    fun `issueContract mints a fresh contract id`() {
        // given: an application / when: a contract is issued twice
        val first = underTest.issueContract(testLeasingApplication().id)
        val second = underTest.issueContract(testLeasingApplication().id)
        // then: each issue gets its own reference
        assertThat(first.value).startsWith("CONTRACT-")
        assertThat(first).isNotEqualTo(second)
    }

    @Test
    fun `revokeContract logs without error`() {
        // given: an issued contract / when-then: revoking it runs without error
        assertThatCode { underTest.revokeContract(ContractId("CONTRACT-1")) }.doesNotThrowAnyException()
    }
}
